### PR TITLE
feat: expose Icon components individually

### PR DESCRIPTION
**Pull request by koba04**
## Overview

We provide many icons through the `Icon` component and its `name` prop like the following.

```jsx
<Icon name="fa-address-book" />
// ↓
<FaAddressBookIcon />
```

This is useful, but it has a drawback, which makes it impossible to do tree-shaking.
So I want to expose the icon components individually.

But it imposes migration costs on us, so I plan to take steps like the following.

1. **[THIS PR]** Add the icon components individually like `<FaAddressBookIcon />`, `<FaCaretDownIcon />` 
1. Migrate the usages of the current Icon component in `smarthr-ui`
1. Add a development warning for the usage of the Icon component
1. Migrate the usages of the current Icon component in applications
1. Remove the current Icon component (**BREAKING CHANGE**)

## What I did

- Add the icon components individually like `FaAddressBookIcon`, `FaCaretDownIcon` 
  - Should I omit the prefix of `Fa`?
  - This also adds the `@deprecated` annotation into `Icon`, `Props`, and `iconMap`.
    - This doesn't affect users of `smarthr-ui`, but it tells us the deprecation when we develop `smarthr-ui` through VSCode.
image

### webpack v5 Production build

- Without Tree Shaking 👉  **202KB**

```jsx
import React from 'react'
import { Icon } from '.'
console.log(<Icon name="fa-address-book" />)
```

- WIth Tree Shaking 👉  **38KB**

```jsx
import React from 'react'
import { FaAddressBookIcon } from '.'
console.log(<FaAddressBookIcon />)
```

### CODE REVIEW

**Comment by wmoai** [comment]
@koba04 Thank you for your great contribution !

 > Should I omit the prefix of Fa?

It is better to be able to recognize the icon that comes from `FontAwesome` so please keep `Fa` prefix !

**Review by wmoai — COMMENTED** [review]
@koba04 Thank you for your great contribution !
I left a comment !

File: src/components/Icon/Icon.tsx [hunk 1]
```diff
@@ -620,16 +626,19 @@ export const iconMap = {
   'fa-yen-sign': FaYenSign,
 }
 
-export const Icon: React.FC<Props> = ({
-  name,
+export type ComponentProps = Omit<Props, 'name'>
+type IconComponentProps = ComponentProps & { Component: React.ComponentType<ComponentProps> }
```
[the remark this answers is not in the document]
Comment (koba04) [reply src/components/Icon/Icon.tsx]: @wmoai Thank you! That's more accurate than `ComponentProps`. I've updated it👍

**Review by wmoai — APPROVED** [review]
Thank you !
LGTM ! :tada:

**Comment by koba04** [comment]
@wmoai oops, I've fixed a type error after your approval. Please approve it again 🙏

**Review by wmoai — DISMISSED** [review]
👍

File: src/components/Icon/index.ts [hunk 2]
```diff
@@ -1,2 +1,2 @@
-export { Icon, iconMap } from './Icon'
+export * from './Icon'
 export type { Props } from './Icon'
```
Comment (negiandleek) [inline src/components/Icon/index.ts:2]: [question]
すべてexportしてると思うんですが、export type してるのって特別な理由があるんでしょうか 🤔

**Review by negiandleek — DISMISSED** [review]
tree-shaking できるようになるのありがたい 🙏

File: src/components/Icon/index.ts (hunk 2 above)
Comment (koba04) [reply src/components/Icon/index.ts:2]: @negiandleek あ、これ最初は個別 export してたときの名残ですね。不要なので消しておきます！あざす！！ 👏

**Review by negiandleek — DISMISSED** [review]
LGTM 🎉

**Review by cidermitaina — APPROVED** [review]

**Pull request merged**